go-controller/pkg/clustermanager/userdefinednetwork/controller_helper.go

```diff
@@ -15,10 +15,25 @@ import (
 	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
 
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/template"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 	utiludn "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util/udn"
 )
 
 func (c *Controller) updateNAD(obj client.Object, namespace string) (*netv1.NetworkAttachmentDefinition, error) {
+	if utiludn.IsPrimaryNetwork(template.GetSpec(obj)) {
```

Comment:
Does it make sense to move the label check under the existing primary-network spec check? (line 54)

Reply:
I'm *assuming* we just want to bail out from this asap ...